src/server/app/util/services/inactivity-email.server.service.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+let _ = require('lodash'),
+	path = require('path'),
+	q = require('q'),
+	deps = require(path.resolve('./src/server/dependencies.js')),
+	handlebars = require('handlebars'),
+	fs = require('fs'),
+	dbs = deps.dbs,
+
+	configc = deps.config,
+	emailService = deps.emailService,
+	logger = deps.logger,
+	User = dbs.admin.model('User');
+
+
+function buildEmailContent(resource, emailTemplateName, config) {
+	let numOfDays = Math.floor((Date.now() - resource.lastLogin)/config.day);
+	let emailData = {
+		appName: configc.app.instanceName,
+		url: configc.app.url.protocol + '://' + configc.app.url.host + ':' + configc.app.url.port + '/#', // message of the day board
+		name: resource.name,
+		date: /* new Date(resource.lastLogin) */ numOfDays
+	};
+	let emailHTML = fs.readFileSync(`./src/server/app/util/templates/${emailTemplateName}-email.server.view.html`, 'utf-8');
+	let template = handlebars.compile(emailHTML);
+	emailHTML = template(emailData);
+	return emailHTML;
+}
+
+/**
+ * alert users who's accounts have been inactive for 30-89 days. Remove accounts that have been inactive for 90+ days
+ */
+module.exports.run = function(config) {
+
+	// search for users inactive for 30 days
+	let query1 = {lastLogin: {$lte: new Date(Date.now() - config.alertInterval[0]), $gt: new Date(Date.now() - config.alertInterval[0] - config.day)}};
+	// search for users inactive for 60 days
+	let query2 = {lastLogin: {$lte: new Date(Date.now() - config.alertInterval[1]), $gt: new Date(Date.now() - config.alertInterval[1] - config.day)}};
+	// search for users inactive for 90 days
+	let query3 = {lastLogin: { $lte: new Date(Date.now() - config.alertInterval[2]), $gt: new Date(Date.now() - config.alertInterval[2] - config.day)}};
+
+	return q.all([User.search(query1), User.search(query2), User.search(query3)])
```

Comment:
Would recommend using a map here as well. So taking the alertQueries array from the previous map, you can do:
`
return q.all(alertQueries.map((query) => User.search(query)))
`